interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -206,6 +273,48 @@ func (udb *userPromptDB) remove(id prompting.IDType) (*Prompt, error) {
 	return prompt, nil
 }
 
+// timeoutCallback is the function which should be called when the expiration
+// timer for the receiving user prompt DB expires. This method should never be
+// called directly outside of a `time.AfterFunc` call which initializes the
+// timer for the user prompt DB when it is first created.
+func (udb *userPromptDB) timeoutCallback(pdb *PromptDB, user uint32) {
+	pdb.mutex.Lock()
+	if pdb.maxIDMmap.IsClosed() {
+		return
+	}
+	// Restart expiration timer while holding the lock, so we don't
+	// overwrite a newly-set activity timeout with an initial timeout.
+	// With the lock held, no activity can occur, so no activity timeout
+	// can be set.
+	if udb.resetExpiration(initialTimeout) {
+		// Timer was active again, suggesting that some activity caused
+		// the timer to be reset at some point between the timer firing
+		// and the lock being released and subsequently acquired by this
+		// function. So reset the timer to activityTimeout, and do not
+		// purge prompts.
+		udb.resetExpiration(activityTimeout)
```

Comment:
this case is not tested

Reply:
It used to be before I lost some commits in a bad force push. Should be fixed now, I cherry picked them.